homework-submission/week2/App.js

```diff
@@ -0,0 +1,127 @@
+const List = ({ description, deadline, className, removeFromState }) => {
+  return (
+    <li className={className} onClick={removeFromState}>
+      {description}, {deadline}
+    </li>
+  );
+};
+
+const UnorderedDynamicList = ({ todos, removeFromState }) => {
+  const processTodos = todos.map(todo => (
+    <List
+      description={todo.description}
+      deadline={todo.deadline}
+      key={todo.id}
+      className={todo.done ? 'finished' : ''}
+      removeFromState={() => {
+        removeFromState(todo.id);
+      }}
+    />
+  ));
+
+  return <ul>{processTodos}</ul>;
+};
+
+class DynamicList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      toDos: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+      newItem: {
+        id: '',
+        description: 'Description',
+        deadline: 'nn',
+        done: false,
+      },
+    };
+  }
+
+  NewListItemDescription = () => {
+    return (
+      <p>
+        <label htmlFor="">Description</label>
+        <input
+          type="text"
+          onChange={event => {
```

Comment:
it's fine to keep a complete function here for _demonstrating_ purposes, but not in a production!
this will be a confusing part for anyone reading the code.

Usually assigning a value will be once and forever, especially when we are dealing with ids, making them very unique to the item and not any other, you are doing that two times first on line 68 and the second on line 86.

and you are doing that on any change so if no changes happens and you wished to have the same exact content of the previous item here is how it goes:

![ezgif com-video-to-gif(2)](https://user-images.githubusercontent.com/25272822/60647526-37a17580-9e3e-11e9-97f2-ff8bf45e78f9.gif)

And note that the console will complain about `Warning: Encountered two children with the same key`

So I would leave this event arrow function here but the id submission in a click event attached to the add button instead!
This will help keeping it:
 1. added once 
 2. submitting as many as wanted on every click since it's a click event 🙂